vortex-layout/src/scan/mod.rs

```diff
@@ -48,6 +53,8 @@ pub struct ScanBuilder<A> {
     /// The executor used to spawn each split task.
     executor: Option<Arc<dyn TaskExecutor>>,
     metrics: VortexMetrics,
+    /// Should we try to prune the file (using stats) on open.
+    file_stats: Option<Arc<[StatsSet]>>,
```

Comment:
I think you could actually create a `FileStatsLayoutReader` and provide a `impl PruningEvaluator`. Then inside `VortexFile::layout_reader()` you can wrap up the root layout with the `FileStatsLayoutReader`, that way the scan never even knows.

Reply:
We agreed this does extra work, so going to add a method on the vortex file itself